cassovary-core/src/main/scala/com/twitter/cassovary/graph/Traverser.scala

```diff
@@ -363,7 +363,7 @@ trait QueueBasedTraverser extends Traverser {
   /**
    * Performs action needed when visiting a node.
    */
-  protected def processNode(node : Node): Node = {
+  protected def processNode[V <: Node](node : V): V = {
```

Comment:
Same problem as above U/W discussion. See if you can simply have processNode(Node) here. Also, seems like the return value is not being used so far, so return Unit.